clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -1006,6 +1036,9 @@ public String toEnrichedRst() {
 
         String lastKeyGroupName = "";
         for (ConfigKey key : sortedConfigs()) {
+            if (key.deprecated) {
+                continue;
```

Comment:
Actually I think #2456 does not update the printed html for marking these as deprecated, so we still need to rebase + merge this PR after that one is merged.